include/envoy/http/header_map.h

```diff
@@ -356,8 +356,8 @@ class HeaderEntry {
  */
 #define DEFINE_INLINE_HEADER(name)                                                                 \
   virtual const HeaderEntry* name() const PURE;                                                    \
-  virtual HeaderEntry* name() PURE;                                                                \
-  virtual HeaderEntry& insert##name() PURE;                                                        \
+  virtual const HeaderEntry& insert##name() PURE;                                                  \
+  virtual void append##name(absl::string_view data, absl::string_view delimiter = ",") PURE;       \
```

Comment:
No default argument for virtual methods: https://google.github.io/styleguide/cppguide.html#Default_Arguments

Reply:
Is there any case in which we don't actually use a `,` delimiter? Wondering if we can just remove this for now?

Edit: I see this is used for manual appending onto an existing header. I think it's fine to have the delimiter be explicit on each call.

Reply:
Yep, there were those 3 cases of not using a `,` delimiter. It's explicit now.

We don't ever do this for non-inline headers, so the `HeaderMapImpl::append(key, value)` method doesn't take a delimiter. Is that clear? Or should I keep methods uniform for consistency?

Reply:
IMO it's fine to have them be different. I would rather have less args for now unless we actually need them.